### PR TITLE
Conflict resolution

### PR DESCRIPTION
**Pull request by ganhongyao**
Changes
- Add conflict resolver. To be treated as a black box with no side effects. Takes in `localModels` and `serverModels` and returns a `ConflictResolution`, the caller will then decide what to do with the result

Tests
- I can write unit tests if needed

### CODE REVIEW

File: Annotato/Annotato/ConflictResolution/ConflictResolver.swift [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            } else if localModel.isDeleted && serverModel.isDeleted {
+                resolution.nonConflictingModels.append(serverModel)
+            } else if localModel == serverModel {
+                resolution.localUpdate.append(serverModel)
+                resolution.nonConflictingModels.append(serverModel)
+            } else {
+                resolution.conflictingModels.append((localModel, serverModel))
+            }
+        }
+    }
```
Comment (sivayogasubramanian) [inline Annotato/Annotato/ConflictResolution/ConflictResolver.swift]: Is it possible to refactor this into different methods for handling the different cases? Or at least we should add some comments explaining why we choose to do certain things.

File: Annotato/Annotato/ConflictResolution/ConflictResolver.swift [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                    resolution.serverUpdate.append(localModel)
+                    resolution.nonConflictingModels.append(localModel)
+                } else {
+                    resolution.localDelete.append(serverModel)
+                    resolution.nonConflictingModels.append(serverModel)
+                }
+            } else if localModel.isDeleted && serverModel.isDeleted {
+                resolution.nonConflictingModels.append(serverModel)
+            } else if localModel == serverModel {
+                resolution.localUpdate.append(serverModel)
```
Comment (sivayogasubramanian) [inline Annotato/Annotato/ConflictResolution/ConflictResolver.swift]: Is this update needed?

Edit: I realized this is to update timestamps.

**Review by sivayogasubramanian — COMMENTED** [review]
I checked through the main part of this PR and the logic seems correct. I didn't really check through the boilerplate hashable and equatable stuff.

**Review by darren2pro — COMMENTED** [review]
Alrights I read through, looks good to me

So right now to integrate with mine, I'll utilize the
```
let conflictResolver = ConflictResolver()
// conflictResolver assign local annotations, remote annotations
let conflictResolution = conflictResolver.resolve()
```

in the document edit view. Somewhere in the view did load function or something, correct

We can discuss tomorrow after your exam

**Comment by ganhongyao** [comment]
@darren2pro specifically like this, you'll need to pass in the models
```
let conflictResolution = ConflictResolver(localModels: localAnnotations, serverModels: serverAnnotations).resolve()
```

**Review by sivayogasubramanian — APPROVED** [review]
LGTM! Thanks

**Pull request merged**